File: hw/src/main/java/com/sumian/app/improve/widget/TabIndicatorItemView.java
package com.sumian.app.improve.widget;

import android.content.Context;
import android.support.annotation.DrawableRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import com.sumian.app.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by sm
 * on 2018/3/6.
 * desc:
 */

public class TabIndicatorItemView extends FrameLayout implements View.OnClickListener {

    @BindView(R.id.tv_tab_text)
    TextView mTvTabText;
    @BindView(R.id.v_tab_dot)
    View mIvTabDot;

    @BindView(R.id.iv_calendar)
    ImageView mIvCalendar;

    private OnSelectTabCallback mOnSelectTabCallback;

    public TabIndicatorItemView(@NonNull Context context) {
        this(context, null);
    }

    public TabIndicatorItemView(@NonNull Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public TabIndicatorItemView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        setOnClickListener(this);
        initView(context);
    }

    private void initView(Context context) {
        ButterKnife.bind(inflate(context, R.layout.hw_lay_tab_dot, this));
        mTvTabText.setVisibility(GONE);
        mIvCalendar.setVisibility(GONE);
        mIvTabDot.setVisibility(GONE);
    }

    public void setOnSelectTabCallback(OnSelectTabCallback onSelectTabCallback) {
        mOnSelectTabCallback = onSelectTabCallback;
    }

    @Override
    public void onClick(View v) {
        if (mOnSelectTabCallback != null) {
            mTvTabText.setActivated(true);
            mOnSelectTabCallback.onSelect(v, mTvTabText.isActivated());
        }
    }

    public void unSelect() {
        mTvTabText.setActivated(false);
    }

    public void select() {
        mTvTabText.setActivated(true);
    }

    public void setIndicatorText(String text) {
        mTvTabText.setText(text);
        mTvTabText.setVisibility(!TextUtils.isEmpty(text) ? VISIBLE : GONE);
    }

    public void setCalendarIcon(@DrawableRes int drawable) {
        mIvCalendar.setImageResource(drawable);
        mIvCalendar.setVisibility(drawable > -1 ? VISIBLE : GONE);
    }

    public void showDot() {
        mIvTabDot.setVisibility(VISIBLE);
    }

    public void hideDot() {
        mIvTabDot.setVisibility(GONE);
    }


    public interface OnSelectTabCallback {

        void onSelect(View v, boolean isSelect);
    }

}

File: app/src/main/java/com/sumian/hw/report/widget/tab/TabIndicatorItemView.kt
package com.sumian.hw.report.widget.tab

import android.annotation.SuppressLint
import android.content.Context
import android.text.TextUtils
import android.util.AttributeSet
import android.view.View
import android.widget.FrameLayout
import com.sumian.common.widget.voice.IVisible
import com.sumian.sd.R
import kotlinx.android.synthetic.main.hw_lay_tab_dot.view.*

/**
 * Created by sm
 * on 2018/3/6.
 * desc:
 */

class TabIndicatorItemView : FrameLayout, View.OnClickListener, IVisible {

    private var mOnSelectTabCallback: OnSelectTabCallback? = null
    private var mEnableTab = false

    constructor(context: Context) : this(context, null)
    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, 0)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : this(context, attrs, defStyleAttr, 0)
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
        val attributes = context.obtainStyledAttributes(attrs, R.styleable.TabIndicatorItemView, defStyleAttr, defStyleRes)

        mEnableTab = attributes.getBoolean(R.styleable.TabIndicatorItemView_enable_tab, false)

        attributes.recycle()
        setOnClickListener(this)
        initView(context)
    }


    private fun initView(context: Context) {
        setPadding(0, resources.getDimensionPixelSize(R.dimen.space_32), 0, resources.getDimensionPixelSize(R.dimen.space_14))
        View.inflate(context, R.layout.hw_lay_tab_dot, this)
        tv_tab_text.visibility = View.GONE
        iv_calendar.visibility = View.GONE
        v_tab_dot.visibility = View.GONE
    }

    fun setOnSelectTabCallback(onSelectTabCallback: OnSelectTabCallback) {
        mOnSelectTabCallback = onSelectTabCallback
    }

    override fun onClick(v: View) {
        tv_tab_text.isActivated = true
        mEnableTab = tv_tab_text.isActivated
        mOnSelectTabCallback?.onSelect(v, tv_tab_text.isActivated)
    }

    fun unSelect() {
        tv_tab_text.isActivated = false
        mEnableTab = tv_tab_text.isActivated
    }

    fun select() {
        tv_tab_text.isActivated = true
        mEnableTab = tv_tab_text.isActivated
    }

    fun setIndicatorText(text: String) {
        tv_tab_text.text = text
        tv_tab_text.visibility = if (!TextUtils.isEmpty(text)) View.VISIBLE else View.GONE
    }

    @SuppressLint("ResourceType")
    fun showCalendar(isShow: Boolean) {
        iv_calendar.visibility = if (isShow) View.VISIBLE else View.GONE
    }

    fun showDot() {
        v_tab_dot.visibility = View.VISIBLE
    }

    fun hideDot() {
        v_tab_dot.visibility = View.GONE
    }

    fun isEnableTab(): Boolean {
        return mEnableTab
    }

    override fun show() {
        visibility = View.VISIBLE
    }

    override fun hide() {
        visibility = View.GONE
    }


    interface OnSelectTabCallback {

        fun onSelect(v: View, isSelect: Boolean)

    }

}
